server/src/test/java/org/candlepin/controller/PoolManagerFunctionalTest.java

```diff
@@ -241,7 +251,7 @@ public void testRegenerateEntitlementCertificatesWithSingleEntitlement()
         throws Exception {
         AutobindData data = AutobindData.create(childVirtSystem).on(new Date())
                 .forProducts(new String [] {provisioning.getId()});
-        this.entitlementCurator.refresh(poolManager.entitleByProducts(data).get(0));
```

Comment:
Why removing refresh on this line? 

The regenerateECAndAssertNotSameCertificates in this test uses the entiteledCertificate that you produced [1] (I saw the same id in the debugging session). By not refreshing, the method regenerateECAndAssertNotSameCertificates will see only the memory image of the created cert, not what has been actualy stored into database.

In case you think refresh really not needed here, then we should remove call to '.get()' as it is superficial without the refresh.

[1]

```
    private Set<EntitlementCertificate> collectEntitlementCertIds(
        Consumer consumer) {
        Set<EntitlementCertificate> ids = Util.newSet();
        for (Entitlement entitlement : consumer.getEntitlements()) {
            for (EntitlementCertificate ec : entitlement.getCertificates()) {
                ids.add(ec);
            }
        }
        return ids;
    }
```